src/common/analytics/Analytics.ts

```diff
@@ -0,0 +1,53 @@
+export const GA_MEASUREMENT_ID = `G-13GYPPVD4Y`;
+
+type GaActionDefault = "page_view";
+type GaAction =
+  | "magic_demo_start"
+  | "magic_demo_issue"
+  | "magic_demo_downloaded"
+  | "magic_demo_file_drop"
+  | "magic_demo_drop_off";
+type GaCategory = "magic_demo";
+
+interface GaEventProps {
+  action: GaAction;
+  category: GaCategory;
+  label?: string;
+  value?: number;
+}
+
+interface GaPageViewProps {
+  action: GaActionDefault;
+}
+
+export const validatePageViewEvent = (event: GaPageViewProps): void => {
+  const { action } = event;
+  if (!action) throw new Error("Action is required");
+  if (action && typeof action !== "string") throw new Error("Action must be a string");
+};
+
+export const gaPageView = (event: GaPageViewProps): void => {
+  validatePageViewEvent(event);
+  const { action } = event;
+  gtag("event", action, {
+    send_to: GA_MEASUREMENT_ID,
+  });
+};
+
+export const validateGaEvent = (event: GaEventProps): void => {
+  const { action, category, label, value } = event;
+  if (!category) throw new Error("Category is required");
+  if (!action) throw new Error("Action is required");
+  if (label && typeof label !== "string") throw new Error("Label must be a string");
+  if (value && typeof value !== "number") throw new Error("Value must be a number");
+};
+
+export const gaEvent = (event: GaEventProps): void => {
+  validateGaEvent(event);
+  const { action, category, label, value } = event;
+  gtag("event", action, {
+    event_category: category,
+    event_label: label,
+    value: value,
+  });
+};
```

Comment:
i was thinking wehther is it too harsh to throw an error if the payload is not correct for both page view and event.

if we let it fail silently, our logs will not have the analytics but the site still can function
but if we let it explode then the whole app will have a hard dependency on google analytics

Reply:
ok

Reply:
👍